packages/gatsby/src/schema/infer/inference-metadata.js

```diff
@@ -181,85 +285,58 @@ const _updateValueDescriptor = (
   }
 
   switch (typeName) {
-    case `object`: {
-      const { props = {} } = typeInfo
-      Object.keys(value).forEach(key => {
-        const v = value[key]
-
-        const [propDescriptor, propDirty] = updateValueDescriptor(
-          {
-            nodeId,
-            key,
-            value: v,
-            operation,
-            descriptor: props[key],
-          },
-          path
-        )
-        props[key] = propDescriptor
-        dirty = dirty || propDirty
-      })
-      typeInfo.props = props
-      break
-    }
-    case `array`: {
-      value.forEach(item => {
-        const [itemDescriptor, itemDirty] = updateValueDescriptor(
-          {
-            nodeId,
-            descriptor: typeInfo.item,
-            operation,
-            value: item,
-            key,
-          },
-          path
-        )
-
-        typeInfo.item = itemDescriptor
-        dirty = dirty || itemDirty
-      })
-      break
-    }
+    // I want to return the func call to use TCO, eslint ignores that case :(
```

Comment:
I believe we [don't have TCO in V8](https://kangax.github.io/compat-table/es6/#test-proper_tail_calls_(tail_call_optimisation)) anyway most likely it will require an [explicit syntax](https://github.com/tc39/proposal-ptc-syntax) when added. So I don't even try optimizing it this way for now %)

Reply:
Hm I have missed this drama. Alright, if tco's are irrelevant then it all doesn't matter (although the separate returns are ugly but meh)